### ISSUE TITLE
Training with images with no label

### ISSUE THREAD

**Issue by seojpark91** [IDEA-Research/DINO#181]
Dear team,
Thanks for your awesome work. I have been experimenting with my custom dataset, and I was wondering whether I can train with some images without boxes/labels. It throws a size mismatch error when concatenating tensors for creating refpoint_embed (in deformable_transformer.py) if an image with no label/ no box is concatenated. The reason for including images without labels/boxes is that there are some objects that can easily be mistaken for an object ( for example, blueberry muffin with chocolate muffin). So I want to give some images with chocolate muffin but without box/label so that the network can learn that. Would that be possible with DINO? If it's possible, how can I modify my target dataset without boxes/labels ?

Thank you!

**Comment by half-truism** [comment]
#38 

It should work out of the box with negative example images (those with no labeled bounding boxes in them). I was able to train with a 1:1 negative to positive imagery ratio. There is likely some issue with the imagery data you are using and not the fact that some imagery does not have labels within it.

"When there is no target in the current sample, DINO should have the ability to classify all targets as no obj, so there will be loss_cls that need to be backward. In the code, It can be seen that the authors have done a lot of clever detailings on this kind of samples to prevent this special case from raising errors in distributed mode."

**Comment by seojpark91** [comment]
Thanks for your comment! I keep getting an error in loss function here:
`indices = self.matcher(outputs_without_aux, targets)`
when calculating cost bbox here
`cost_bbox = torch.cdist(out_bbox, tgt_bbox, p=1)`

![Screen Shot 2023-04-18 at 9 13 52 AM](https://user-images.githubusercontent.com/41909123/232637054-9524d94c-0761-4dcb-8660-d6cf84887c3f.png)

Is there specific input format required for negative samples? I was giving empty bbox (tensor with length 0) and empty label (also tensor with length 0 ) but it throws error like above.

**Comment by seojpark91** [comment]
Although I gave empty bboxes with shape (0,4) and labels with shape (0), it would throw an error in `prepare_for_cdn` function here.
```
known = [(torch.ones_like(t["labels"])).cuda() for t in targets]
batch_size = len(known)
known_num = [sum(k) for k in known]
if int(max(known_num)) == 0:
```
since `known_num` has length 0, it throws an error. Should I replace no label with background label index?
